Dynatrace/dynatrace-operator
# Deprecate old csi tables

## Description

This PR resolves [issue](https://dt-rnd.atlassian.net/browse/K8S-9020) 

## How can this be tested?

1. Deploy the operator to your cluster (version 1.0 or early).
2. create Dynakybe, and make sure `csi.sqlite` has old tables!
```
$ kubectl cp dynatrace-oneagent-csi-driver-5fw8l:/data/csi.db csi.sqlite
tar: Removing leading `/' from member names
$ sqlite3 csi.sqlite
SQLite version 3.43.2 2023-10-10 13:08:14
Enter ".help" for usage hints.
sqlite> .tables
app_mounts       migrations       tenant_configs
code_modules     os_mounts        volume_meta
dynakubes        osagent_volumes  volumes
sqlite>
```

```
sqlite> select * from migrations;
SCHEMA_INIT
202403041200
sqlite>
```
3. Then upgrade to the current version.
4. Copy DB to your local machine: `kubectl cp dynatrace-oneagent-csi-driver-<hash>:/data/csi.db csi.sqlite`
5. Run `.tables`:
```
sqlite> .tables
app_mounts      migrations      tenant_configs
code_modules    os_mounts       volume_meta
```

sqlite> select * from migrations;
SCHEMA_INIT
202403041200
202406061200 <----- NEW MIGRATION
sqlite>

and not presented tables: `"dynakubes", "volumes", "osagent_volumes"`
